app/cdap/components/WranglerGrid/SelectColumnPanel/ColumnsList/index.tsx

```diff
@@ -0,0 +1,218 @@
+    const multiSelect: IMultipleSelectedFunctionDetail[] = MULTI_SELECTION_COLUMN.filter(
+      (functionDetail: IMultipleSelectedFunctionDetail) =>
+        functionDetail.value === transformationName && functionDetail.isMoreThanTwo
+    );
+    if (selectedColumns?.length === 0 || selectedColumns?.length < 2) {
+      return false;
+    } else if (multiSelect.length) {
+      return false;
+    } else {
+      return true;
+    }
+  };
+
+  const handleSearch = (event: React.ChangeEvent<HTMLInputElement>) => {
+    if (event.target.value) {
+      const columnValue: IHeaderNamesList[] = filteredColumnsOnType.length
+        ? filteredColumnsOnType.filter((columnDetail: IHeaderNamesList) =>
+            columnDetail.label.toLowerCase().includes(event.target.value.toLowerCase())
+          )
+        : [];
+      columnValue?.length ? setColumns(columnValue) : setColumns([]);
+    } else {
+      setColumns(filteredColumnsOnType);
+    }
+  };
+
+  const handleFocus = () => {
+    ref?.current?.focus();
+  };
+
+  return (
+    <SelectColumnWrapper data-testid="select-column-list-parent">
+      {columnsAsPerType.length === 0 && (
+        <FlexWrapper>
+          <CenterAlignBox>
+            {NoDataSVG}
+            <SubHeadBoldFont component="p" data-testid="no-column-title">
+              {T.translate(`${SELECT_COLUMN_LIST_PREFIX}.noColumns`)}
+            </SubHeadBoldFont>
+            <NormalFont component="p" data-testid="no-column-subTitle">
+              {T.translate(`${SELECT_COLUMN_LIST_PREFIX}.noMatchColumnDatatype`)}
+            </NormalFont>
+          </CenterAlignBox>
+        </FlexWrapper>
+      )}
+      {columnsAsPerType.length > 0 && (
+        <>
+          <SelectColumnInnerWrapper>
+            <SelectedColumnCountWidget selectedColumnsCount={selectedColumnsCount} />
+            <SelectColumnSearchBox>
+              <SelectColumnSearchInput
+                data-testid="input-search-id"
+                onChange={handleSearch}
+                ref={ref}
+              />
+              <SearchIconButton onClick={handleFocus} data-testid="click-handle-focus">
+                <SearchIconComponent />
+              </SearchIconButton>
+            </SelectColumnSearchBox>
+          </SelectColumnInnerWrapper>
+          <ColumnTable
+            dataQualityValue={dataQuality}
+            onSingleSelection={onSingleSelection}
+            handleDisableCheckbox={handleDisableCheckbox}
+            onMultipleSelection={onMultipleSelection}
+            columns={columnsAsPerType.length === 0 ? [] : columns}
+            transformationDataType={transformationDataType}
+            isSingleSelection={isSingleSelection}
+            selectedColumns={selectedColumns}
+            totalColumnCount={columnsAsPerType?.length}
+            setSelectedColumns={setSelectedColumns}
+            transformationName={transformationName}
+          />
+        </>
+      )}
+    </SelectColumnWrapper>
+  );
+}
```

Comment:
as the total number of lines in this component is exceeding 150-200 rule
can we move the styled-components to a separate file called styles.ts and import here

Reply:
Done